src/runtime/HalideRuntime.h

```diff
@@ -130,13 +142,17 @@ extern int halide_do_task(void *user_context, halide_task_t f, int idx,
                           uint8_t *closure);
 //@}
 
-/** Spawn a thread, independent of halide's thread pool. */
-extern void halide_spawn_thread(void *user_context, void (*f)(void *), void *closure);
+struct halide_thread;
+
+/** Spawn a thread. Returns a handle to the thread for the purposes of joining it. */
```

Comment:
This should make it clear the join is required and will cleanup any resources associated with the thread.